app/renderer/lib/appium-client.js

```diff
@@ -221,90 +227,79 @@ export default class AppiumClient {
     return {windowSize, windowSizeError};
   }
 
-  async getContextUpdate () {
-    let contexts,
-        contextsError,
-        currentContext,
-        currentContextError,
-        pixelRatio,
-        platformName,
-        statBarHeight,
-        viewportRect,
-        webViewPosition;
+  // Retrieve all detected contexts, as well as the current context
+  // If retrieval of either one fails, return the error(s)
+  // Additionally, if webview is used, adjust the found element positions to fit screenshot
+  // Only called while in hybrid mode
+  async getContextUpdate ({ windowSize }) {
+    let contexts, contextsError, currentContext, currentContextError, webviewTopOffset;
+    let webviewLeftOffset = 0;
+
     if (!await this.hasContextsCommand()) {
       return {currentContext: null, contexts: []};
     }
 
+    // First get the current context (or the error, if one appears)
     try {
       currentContext = await this.driver.getContext();
     } catch (e) {
       currentContextError = e;
     }
 
-    // Note: These methods need to be executed in the native context because ChromeDriver behaves differently
+    // The retrieval of all contexts and webview position adjustments require some native context use
     if (currentContext !== NATIVE_APP) {
       await this.driver.switchContext(NATIVE_APP);
     }
 
-    ({platformName, pixelRatio, statBarHeight, viewportRect} = await this.driver.getSession());
+    const { platformName, statBarHeight, viewportRect } = await this.driver.getSession();
```

Comment:
are are looking forward to remove this information from GET /session endpoint completely as it is not a part of W3C standard. Consider using extension calls instead

Reply:
Good call, I'll see what I can do.

Reply:
I assume `platformName` can still remain here, since it is part of the standard?
At the same time, according to UiA2 docs, it does not enforce the value to be `Android`, so maybe it's better to check `appium:automationName` instead. But I've never gotten Espresso to work, so I won't know how that behaves...

Reply:
I've now changed both `statBarHeight` and `viewportRect` to be retrieved using mobile extensions. In case the driver is too old and does not support these extensions, it will fallback to the /session endpoint.